src/lib/teuthologyAPI.ts

```diff
@@ -7,28 +7,34 @@ const TEUTHOLOGY_API_SERVER =
     import.meta.env.VITE_TEUTHOLOGY_API || "";
 const GH_USER_COOKIE = "GH_USER";
 
+function getURL(relativeURL: URL|string): string {
+    if ( ! TEUTHOLOGY_API_SERVER ) return relativeURL.toString();
+    return new URL(relativeURL, TEUTHOLOGY_API_SERVER).toString();
+}
+
 function useLogin() {
-    const url =  new URL("/login/", TEUTHOLOGY_API_SERVER).href;
-    window.location.href = url;
+    const url = getURL("/login/");
+    if ( url ) window.location.href = url;
```

Comment:
Should we check if `url !== "/login/"` because, if `TEUTHOLOGY_API_SERVER` is empty, `getURL` would return "/login/" which would redirect to pulpito-ng's /login?

Reply:
When this is paired with the changes to the `Login` component in this PR, we `return null` before `useLogin` would ever be called.

Actually, now that I look more closely, `useLogin` and `useLogout` aren't actually hooks an so shouldn't have `use*` names.

Reply:
I see your point - we won't ever fall into that use-case with current logic.

> Actually, now that I look more closely, useLogin and useLogout aren't actually hooks an so shouldn't have use* names.

Right they shouldn't, thanks for renaming them!